File: msg-key-phrases/src/main/java/dev/number6/keyphrases/dagger/ChannelMessagesHandlerModule.java
package dev.number6.keyphrases.dagger;

import dev.number6.comprehend.results.PresentableKeyPhrasesResults;
import dev.number6.message.ChannelMessagesComprehensionHandler;
import dev.number6.message.ChannelMessagesHandler;
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction;
import dev.number6.message.ComprehensionResultsConsumer;
import dagger.Module;
import dagger.Provides;

@Module
public class ChannelMessagesHandlerModule {

    @Provides
    public ChannelMessagesHandler providesMessageComprehension(ChannelMessagesToComprehensionResultsFunction<PresentableKeyPhrasesResults> channelMessagesToComprehensionResultsFunction,
                                                               ComprehensionResultsConsumer<PresentableKeyPhrasesResults> comprehensionResultsConsumer) {
        return new ChannelMessagesComprehensionHandler<>(channelMessagesToComprehensionResultsFunction, comprehensionResultsConsumer);
    }
}

File: msg-key-phrases/src/main/kotlin/dev/number6/keyphrases/dagger/ChannelMessagesHandlerModule.kt
package dev.number6.keyphrases.dagger

import dagger.Module
import dagger.Provides
import dev.number6.comprehend.results.PresentableKeyPhrasesResults
import dev.number6.message.ChannelMessagesComprehensionHandler
import dev.number6.message.ChannelMessagesHandler
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction
import dev.number6.message.ComprehensionResultsConsumer

@Module
class ChannelMessagesHandlerModule {
    @Provides
    fun providesMessageComprehension(channelMessagesToComprehensionResultsFunction: ChannelMessagesToComprehensionResultsFunction<PresentableKeyPhrasesResults>,
                                     comprehensionResultsConsumer: ComprehensionResultsConsumer<PresentableKeyPhrasesResults>): ChannelMessagesHandler {
        return ChannelMessagesComprehensionHandler<PresentableKeyPhrasesResults>(channelMessagesToComprehensionResultsFunction, comprehensionResultsConsumer)
    }
}